components/clp-py-utils/clp_py_utils/s3_utils.py

```diff
@@ -1,18 +1,208 @@
+    """
+    env_vars: Dict[str, str] = None
+    auth = config.aws_authentication
+
+    if AwsAuthType.profile == auth.type:
+        aws_credentials: S3Credentials = _get_session_credentials(auth.profile)
+        if aws_credentials is None:
+            raise ValueError(f"Failed to authenticate with profile {auth.profile}")
+
+        env_vars = {
+            "AWS_ACCESS_KEY_ID": aws_credentials.access_key_id,
+            "AWS_SECRET_ACCESS_KEY": aws_credentials.secret_access_key,
+        }
+        aws_session_token = aws_credentials.session_token
+        if aws_session_token is not None:
+            env_vars["AWS_SESSION_TOKEN"] = aws_session_token
+        return env_vars
+
+    elif AwsAuthType.credentials == auth.type:
+        credentials = auth.credentials
+        env_vars = {
+            "AWS_ACCESS_KEY_ID": credentials.access_key_id,
+            "AWS_SECRET_ACCESS_KEY": credentials.secret_access_key,
+        }
+        if credentials.session_token:
+            env_vars["AWS_SESSION_TOKEN"] = credentials.session_token
+        return env_vars
+
+    elif AwsAuthType.env_vars == auth.type:
+        # Environmental variables are already set
+        return {}
+    elif AwsAuthType.ec2 == auth.type:
+        aws_credentials: S3Credentials = _get_session_credentials()
+        if aws_credentials is None:
+            raise ValueError(f"Failed to authenticate with EC2 metadata.")
+
+        env_vars = {
+            "AWS_ACCESS_KEY_ID": aws_credentials.access_key_id,
+            "AWS_SECRET_ACCESS_KEY": aws_credentials.secret_access_key,
+        }
+        if aws_credentials.session_token:
+            env_vars["AWS_SESSION_TOKEN"] = aws_credentials.session_token
+        return env_vars
+
+    else:
+        raise ValueError(f"Unsupported authentication type: {auth.type}")
+
+
+def generate_container_auth_options(
+    clp_config: CLPConfig, component_type: str
+) -> Tuple[bool, List[str]]:
+    """
+    Generates Docker container authentication options for AWS S3 access based on the given type.
+    Handles authentication methods that require extra configuration (profile, env_vars).
+
+    :param clp_config: CLPConfig containing storage configurations.
+    :param type: Type of calling container (compression, log_viewer, or query).
+    :return: Tuple of (whether aws config mount is needed, credential env_vars to set).
+    :raises: ValueError if environment variables are not set correctly.
+    """
+    if component_type not in [
+        COMPRESSION_SCHEDULER_COMPONENT_NAME,
+        COMPRESSION_WORKER_COMPONENT_NAME,
+        LOG_VIEWER_WEBUI_COMPONENT_NAME,
+        QUERY_SCHEDULER_COMPONENT_NAME,
+        QUERY_WORKER_COMPONENT_NAME,
+    ]:
+        raise ValueError(f"Component type {component_type} is not valid.")
+
+    storages_by_component_type = []
+    if (
+        COMPRESSION_SCHEDULER_COMPONENT_NAME == component_type
+        or COMPRESSION_WORKER_COMPONENT_NAME == component_type
+    ):
```

Comment:
nit: Let's use what's currently used in start-clp.py
```suggestion
    if component_type in (
        COMPRESSION_SCHEDULER_COMPONENT_NAME, 
        COMPRESSION_WORKER_COMPONENT_NAME
    ):
```